### PR TITLE
URL Param + Variadic route doc

### PR DESCRIPTION
**Pull request by dlepaux**

### CODE REVIEW

File: docs/route.md [hunk 1]
```diff
@@ -319,6 +319,14 @@ m.route(document.body, "/edit/pictures/image.jpg", {
 })
 ```
 
+For isomorphic / universal javascript app, an url param and a variadic route combined is very usefull to display custom 404 error page.
+
+In a case of 404 Not Found error, Express or similar send back the custom page to client. When Mithril is loaded, it will redirect client to the default route because it can't know that route.
```
Comment (tivac) [inline docs/route.md]: I'd change "Express or similar" to "the server".

File: docs/route.md [hunk 2]
```diff
@@ -319,6 +319,14 @@ m.route(document.body, "/edit/pictures/image.jpg", {
 })
 ```
 
+For isomorphic / universal javascript app, an url param and a variadic route combined is very usefull to display custom 404 error page.
```
Comment (tivac) [inline docs/route.md]: This should have its own heading like "Handling 404s"

File: docs/route.md [hunk 3]
```diff
@@ -319,6 +319,14 @@ m.route(document.body, "/edit/pictures/image.jpg", {
...
+
+In a case of 404 Not Found error, Express or similar send back the custom page to client. When Mithril is loaded, it will redirect client to the default route because it can't know that route.
+
+```javascript
+"/:404...": errorPageComponent
```
Comment (tivac) [inline docs/route.md]: This should be a complete invocation of `m.route()` instead of a snippet.

**Comment by dlepaux** [comment]
@tivac Update done. It's a pleasure.

**Pull request merged**